File: AdminPanel/AdminPanel.Web/Modules/Conne/Category/CategoryColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel.Conne.Columns;

[ColumnsScript("Conne.Category")]
[BasedOnRow(typeof(CategoryRow), CheckNames = true)]
public class CategoryColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public string Photo { get; set; }
}

File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/Conne/CategoryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CategoryRow } from "./CategoryRow";

export interface CategoryColumns {
    Id: Column<CategoryRow>;
    Name: Column<CategoryRow>;
    Description: Column<CategoryRow>;
    Photo: Column<CategoryRow>;
}

export class CategoryColumns extends ColumnsBase<CategoryRow> {
    static readonly columnsKey = 'Conne.Category';
    static readonly Fields = fieldsProxy<CategoryColumns>();
}